--- ide/lazarusmanager.pas      2013-06-02 21:03:48.000000000 +0000
+++ ide/lazarusmanager.pas.new  2014-03-07 19:03:04.984732503 +0000
@@ -152,9 +152,9 @@
   BackupFilename: String;
   CurFilename: String;
 begin
-  NewFilename:=AppendPathDelim(Directory)+'lazarus.new'+GetExeExt;
-  BackupFilename:=AppendPathDelim(Directory)+'lazarus.old'+GetExeExt;
-  CurFilename:=AppendPathDelim(Directory)+'lazarus'+GetExeExt;
+  NewFilename:=AppendPathDelim(Directory)+GetBaseLazarusExeName+'.new'+GetExeExt;
+  BackupFilename:=AppendPathDelim(Directory)+GetBaseLazarusExeName+'.old'+GetExeExt;
+  CurFilename:=AppendPathDelim(Directory)+GetBaseLazarusExeName+GetExeExt;
   if FileExistsUTF8(NewFileName) then
   begin
     if FileExistsUTF8(CurFilename) then
@@ -304,8 +304,10 @@
     if (RenameLazarusExecutable(DefaultDir)=mrOK)
       and (RenameLazarusExecutable(CustomDir)=mrOK) then
     begin
-      DefaultExe:=DefaultDir+'lazarus'+GetExeExt;
-      CustomExe:=CustomDir+'lazarus'+GetExeExt;
+      DefaultExe:=DefaultDir+GetBaseLazarusExeName+GetExeExt;
+      CustomExe:=CustomDir+GetBaseLazarusExeName+GetExeExt;
+      debugln('Lazarus default exe is: ' + DefaultDir+GetBaseLazarusExeName+GetExeExt);
+      debugln('Lazarus custom exe is: ' + CustomDir+GetBaseLazarusExeName+GetExeExt);
       if FileExistsUTF8(DefaultExe) then begin
         if FileExistsUTF8(CustomExe) then begin
           // both exist
--- ide/lazconf.pp      2014-03-07 18:45:45.416608740 +0000
+++ ide/lazconf.pp.new  2014-03-07 18:47:21.349801356 +0000
@@ -133,6 +133,12 @@
 procedure AddFilenameToList(List: TStrings; const Filename: string;
   SkipEmpty: boolean = true);
 
+type
+  TDesktopEnv = (deUnknow, deGnome, deKDE, deMacOSX, deWindows);
+
+function GetDesktopEnv: TDesktopEnv;
+function GetBaseLazarusExeName: string;
+
 const
   EmptyLine = LineEnding + LineEnding;
   EndOfLine: shortstring = LineEnding;
@@ -157,6 +163,40 @@
 
 {$I lazconf.inc}
 
+function GetDesktopEnv: TDesktopEnv;
+begin
+  Result := deUnknow; //unrecognized desktop environment
+  {$IFDEF MSWINDOWS}
+    Result := deWindows;
+  {$ENDIF}
+  {$IFDEF DARWIN}
+    Result := deMacOSX;
+  {$ENDIF}
+  //if this Gnome specific env variable is set, probably the user is running Gnome
+  if GetEnvironmentVariable('GNOME_DESKTOP_SESSION_ID') <> '' then
+    Result := deGnome;
+  //if this KDE specific env variable is set, probably the user is running KDE
+  if GetEnvironmentVariable('KDE_FULL_SESSION') <> '' then
+    Result := deKDE;
+  //if gnome or kde is explicitely declared in DESKTOP_SESSION env variable,
+  //override previously assumed result
+  if GetEnvironmentVariable('DESKTOP_SESSION') = 'gnome' then
+    Result := deGnome
+  else
+    if Pos('kde', Getenvironmentvariable('DESKTOP_SESSION')) <> 0 then
+      Result := deKDE;
+end;
+
+function GetBaseLazarusExeName: string;
+begin
+  Result:='lazarus';
+  if GetDesktopEnv = deKDE then
+    Result:='lazarus-qt'
+  else
+    if (GetDesktopEnv = deUnknow) or (GetDesktopEnv = deGnome) then
+      Result:='lazarus-gtk';
+end;
+
 procedure AddFilenameToList(List: TStrings; const Filename: string;
   SkipEmpty: boolean);
 var
@@ -170,7 +210,7 @@
 
 function GetLazarusApplicationName: string;
 begin
-  Result := 'lazarus';
+  Result := GetBaseLazarusExeName;
 end;
 
 procedure LazConfSubstituteMacros(var s: string);
