--- gpm-1.20.6.orig/src/report.c	2009-02-09 10:58:53.000000000 +0100
+++ gpm-1.20.6/src/report.c	2009-11-16 02:11:55.000000000 +0100
@@ -48,15 +48,15 @@
  *
  * Startup Mode:
  *    debug    :  - (ignore)
- *    info     :  syslog/stdout
- *    warn/err :  syslog/stderr
+ *    info     :  syslog
+ *    warn/err :  syslog
  *    oops     :  syslog/stderr  [exit]
  *
  * Running Mode: (daemon)
  *    debug    :  - (ignore)
  *    info     :  syslog
- *    warn     :  syslog+system console
- *    err      :  syslog+system console+current console
+ *    warn     :  syslog
+ *    err      :  syslog
  *    oops     :  syslog/stderr  [_exit]
  *
  * Debug Mode  :
@@ -92,9 +92,7 @@
                syslog(LOG_INFO | LOG_USER, GPM_STRING_INFO);
                vsyslog(LOG_INFO | LOG_USER, text, ap2);
 #else
-               fprintf(stderr,GPM_STRING_INFO);
-               vfprintf(stderr,text,ap);
-               fprintf(stderr,"\n");
+
 #endif
                break;
 
@@ -103,9 +101,7 @@
                syslog(LOG_DAEMON | LOG_WARNING, GPM_STRING_WARN);
                vsyslog(LOG_DAEMON | LOG_WARNING, text, ap2);
 #else
-               fprintf(stderr,GPM_STRING_WARN);
-               vfprintf(stderr,text,ap);
-               fprintf(stderr,"\n");
+               
 #endif
                break;
 
@@ -114,9 +110,7 @@
                syslog(LOG_DAEMON | LOG_ERR, GPM_STRING_ERR);
                vsyslog(LOG_DAEMON | LOG_ERR, text, ap2);
 #else
-               fprintf(stderr,GPM_STRING_ERR);
-               vfprintf(stderr,text,ap);
-               fprintf(stderr,"\n");
+               
 #endif
                break;
 
@@ -149,12 +143,7 @@
                syslog(LOG_DAEMON | LOG_WARNING, GPM_STRING_WARN);
                vsyslog(LOG_DAEMON | LOG_WARNING, text, ap2);
 #else
-               if((console = fopen(GPM_SYS_CONSOLE,"a")) != NULL) {
-                  fprintf(console,GPM_STRING_WARN);
-                  vfprintf(console,text,ap);
-                  fprintf(console,"\n");
-                  fclose(console);
-               }   
+                  
 #endif
                break;
  
@@ -163,19 +152,7 @@
                syslog(LOG_DAEMON | LOG_ERR, GPM_STRING_ERR);
                vsyslog(LOG_DAEMON | LOG_ERR, text, ap2);
 #else
-               if((console = fopen(GPM_SYS_CONSOLE,"a")) != NULL) {
-                  fprintf(console,GPM_STRING_ERR);
-                  vfprintf(console,text,ap);
-                  fprintf(console,"\n");
-                  fclose(console);
-               }
-
-               if((console = fopen(option.consolename,"a")) != NULL) {
-                  fprintf(console,GPM_STRING_ERR);
-                  vfprintf(console,text,ap3);
-                  fprintf(console,"\n");
-                  fclose(console);
-               }
+
 #endif
                break;
 
