--- a/config/modules/espeak-mbrola-generic.conf.orig	2011-04-27 12:01:16.000000000 +0200
+++ b/config/modules/espeak-mbrola-generic.conf	2012-03-21 17:25:44.066763025 +0100
@@ -18,7 +18,7 @@
 # can modify this value, see other parameters).
 # The command can be split into more lines, if necessary, using '\'.
 GenericExecuteSynth \
-"echo \'$DATA\' | espeak -v mb-$VOICE $PUNCT -s $RATE -a $VOLUME | mbrola -v $VOLUME -e /usr/share/mbrola/$VOICE/$VOICE - -.au | $PLAY_COMMAND
+"echo \'$DATA\' | espeak -v mb-$VOICE -s $RATE -p $PITCH $PUNCT -q --stdin --pho | mbrola -v $VOLUME -e /usr/share/mbrola/$VOICE/$VOICE - -.au | $PLAY_COMMAND
 
 # The following three items control punctuation levels None, Some, and All.
 # Each of these values will be substituted into the $PUNCT variable depending
