diff --git a/mail/installer/Makefile.in b/mail/installer/Makefile.in
--- a/mail/installer/Makefile.in
+++ b/mail/installer/Makefile.in
@@ -104,16 +104,18 @@ MOZ_PKG_MAC_EXTRA=--symlink "/Applicatio
 endif
 
 NON_OMNIJAR_FILES = defaults/messenger/mailViews.dat
 
 ifndef LIBXUL_SDK
 INSTALL_SDK = 1
 endif
 
+GENERATE_CACHE=1
+
 include $(MOZILLA_SRCDIR)/toolkit/mozapps/installer/packager.mk
 
 ifeq (Darwin, $(OS_ARCH))
 BINPATH = $(_BINPATH)
 DEFINES += -DAPPNAME=$(_APPNAME)
 else
 BINPATH = bin
 endif

