--- a/spack-3.1/src/spackpkg	2011-06-02 16:23:56.000000000 +0200
+++ b/spack-3.1/src/spackpkg	2011-07-10 10:57:11.941981016 +0200
@@ -27,11 +27,11 @@
 deadwarn () {
     # Warn about broken symlinks
     # $f CURRENT WORKING DIRECTORY
-    echo "Checking the package for dead symbolic links..."
+    echo "Vérification des liens symboliques morts..."
     find * -type l -exec stat -c "%N" {} + | symresolv "$1"| while read ln; do
         target="$(echo "$ln" | cut -f 2)"
         [ -e "$target"  ] && continue
-        growl 0 "The \"$(echo "$ln" | cut -f 1)\" symbolic link is broken."
+        growl 0 "Le lien symbolique \"$(echo "$ln" | cut -f 1)\" est cassé."
     done
 }
 
@@ -39,14 +39,14 @@
     # Warn about empty files
     # $f
     local type="-type f -size"
-    echo "Checking the package for empty files..."
+    echo "Vérification des fichiers vides..."
     # True empty files...
-    find * $type 0c | awk '{print "(W) The \""$0"\" regular file is empty."}' >&2
+    find * $type 0c | awk '{print "(A) Le fichier \""$0"\" est vide."}' >&2
     # Compressed empty files (check integrity first)...
     find * $type +0c -name '*.gz' -exec gzip -t  {} + 2>&1 \
-           | awk -F: '/./{gsub(/^ |$/,"\"",$2); print "(W) The "$2" gzipped file is corrupted."}' >&2
+           | awk -F: '/./{gsub(/^ |$/,"\"",$2); print "(A) Le fichier compressé "$2" est corrompu."}' >&2
     find * $type +0c -name '*.gz' -exec gzip -l -q {} + 2>/dev/null \
-           | awk '($2 == 0){print "(W) The \""$4".gz\" gzipped file is empty."}' >&2
+           | awk '($2 == 0){print "(A) Le fichier compressé \""$4".gz\" est vide."}' >&2
 }
 
 explode () {
@@ -55,11 +55,11 @@
     local fmt=$(echo "$2" | sed 's/.*\.t\([gx]\)z$/\1/;/^[gx]$/!d;s/g/z/') 
     local file="$(cd "$(dirname "$2")"; pwd -P)/$(basename "$2")"
     local tmp="$1/$(basename "$2" | sed 's/t[gx]z$/tmpXXXXXX/')"
-    [ "$fmt" -a -f "$file" ] || growl 1 "The \"$file\" does not exist or is not a Slackware(R) package."
-    tmp="$(mktemp -d "$tmp" 2>/dev/null)" || growl 1 "Can't create a secure temporary directory."
+    [ "$fmt" -a -f "$file" ] || growl 1 "\"$file\" n'existe pas ou n'est pas un paquet Slackware."
+    tmp="$(mktemp -d "$tmp" 2>/dev/null)" || growl 1 "Impossible de créer un répertoire temporaire sécurisé."
     ( cd "$tmp" || exit 1
       tar xf$fmt "$file" --same-permissions --same-owner
-    ) || growl 1 "Extraction failed."
+    ) || growl 1 "Échec de l'extraction."
     echo "$tmp"
 }
 
@@ -83,7 +83,7 @@
     local err=$1 pr='(E)'; shift 1
     if [ "$1" ]; then
         # If there is a second argument, we print arguments with 'echo'.
-        [ $err = 0 ] && pr='(W)'
+        [ $err = 0 ] && pr='(A)'
         echo "$pr $*" 1>&2
     else
         # Otherwise we use 'cat' and print stdin.
@@ -96,20 +96,20 @@
     # Manage the installation-related files.
     # $f SANCTUARY
     local tmp="$1"
-    echo "Looking for installation files..."
+    echo "Recherche de fichiers d'installation..."
     if [ -s ./about.txt ]; then
-        echo '    found "about.txt".' # native fashion
+        echo '    "about.txt" détecté.' # native fashion
     elif [ -s ./install/slack-desc ]; then
-        echo '    found "install/slack-desc", move it as "about.txt".' # Slackware compatibility.
+        echo '    "install/slack-desc" détecté, renommage en "about.txt".' # Slackware compatibility.
         mv ./install/slack-desc ./about.txt
     else
-        growl 0 "No description for this package."
+        growl 0 "Il n'y a pas de description pour ce paquet."
         touch ./about.txt # create it to avoid errors in makefiles()
     fi
     if [ -s ./post-install.sh ]; then
-        echo '    found "post-install.sh".' # native fashion
+        echo '    "post-install.sh" détecté.' # native fashion
     elif [ -s ./install/doinst.sh ]; then
-        echo '    found "install/doinst.sh", move it as "post-install.sh".' # Slackware compatibility.
+        echo '    "install/doinst.sh" détecté, renommage en "post-install.sh".' # Slackware compatibility.
         mv ./install/doinst.sh ./post-install.sh
     fi
     rm -rf ./install/ # from now on useless.
@@ -125,17 +125,17 @@
     # We archive and compress the files
     ext=$(basename "$compress" | cut -c -2)
     echo "================================================================================"
-    echo "Making the \"files.$ext\" sub-archive..."
+    echo "Création de la sous-archive \"files.$ext\"..."
     echo "================================================================================"
     getfiles | cpio --quiet -o -v -H newc 2>&3 | $compress > "$tmp/files.$ext" # CRC useless here...
     install -m 0644 ./about.txt "$tmp"
     # We create the final package...
     echo "================================================================================"
-    echo "Making the \"$target\" final archive..." 
+    echo "Création de l'archive finale \"$target\"..." 
     echo "================================================================================"
     ( cd $tmp &&  ls -1 | cpio --quiet -o -v -H crc 2>&3 >"$target" ) || exit 1
     echo ""
-    echo "Removing the temporary stuff..."
+    echo "Nettoyage des fichiers temporaires..."
     rm -rf "$tmp"
 }
 
@@ -184,7 +184,7 @@
     elif [ -f "./$target" ]; then
         cat "./$target" >&4
     fi
-    echo "$add the symbolic links in the post-installation script..."
+    echo "Ajout des liens symboliques dans la post-installation..."
     find * -type l | while read from; do
         echo "    store $from" 
         dir=$(dirname "$from"); to=$(readlink "$from"); from=$(basename "$from")
@@ -203,39 +203,38 @@
     local out=1 code="$1"
     [ $code -eq 0 ] || out=2
     cat >&$out << HELP
-spackpkg - produce a Spack package.
+spackpkg - crée un paquet Spack.
 
-    spackpkg makes CPIO-based packages of the Spack format. This one is fully
-    compatible with Slackware(R) but requires spackadd/spackrm in order to be
-    managed.
-
-    Usage: spackpkg <directory|T[GX]Z Slackware(R) package> [output] [options]
-
-    Options:
-
-        -c, --clean-up          ---     remove the directory once it's packaged.
-        -h, --help              ---     print this help on stdout then exit.
-        -p, --prepend-symlinks  ---     prepend symlinks restitution code part
-                                        to the "post-install.sh" script instead
-                                        of append it.
-        -q, --quiet             ---     mute the messages sent to stdout.
-        -r, --reset-permissions ---     reset the directory modes to "0755"
-                                        and attribute the tree to the superuser
-                                        before packaging.
-        -z, --gzip              ---     use "gzip" to compress the files instead
-                                        of "xz". This is the default when "xz"
-                                        is not found on the system.
-
-    Environment:
-
-        TMPDIR                  ---     define another directory instead of
-                                        "/tmp", where intermediate steps of the
-                                        packaging process will be performed.
-
-
-    Bug reports, suggestions, feedbacks, questions, and so on, should be sent to
-    Seb (author/maintainer, the license -- MIT/X11 -- is at the head of the
-    script) here: <sbb at tuxfamily dot org>.
+    'spackpkg' crée des paquets basés sur CPIO au format Spack, lequel est 100%
+    compatible avec le format Slackware mais requiert les outils 'spackadd' et
+    'spackrm' pour en gérer l'installation.
+
+    Utilisaion: spackpkg [répertoire|paquet Slackware] [sortie] [options]
+
+    Options :
+
+        -c, --clean-up          Supprime le réperoire une fois empaqueté.
+        -h, --help              Affiche cette aide sur la sortie et quitte.
+        -p, --prepend-symlinks  Injecte le code de création des liens en
+                                début de script plutôt qu'à la fin.
+        -q, --quiet             Affichage moins verbeux.
+        -r, --reset-permissions Réaffecte les droits des répertoires à "0755"
+                                et réattribue l'appartenance de l'arborescence
+                                au super-utilisateur avant d'empaqueter.
+        -z, --gzip              Utilise 'gzip' pour compresser les fichiers en
+                                lieu et place de 'xz'. Comportement par défaut
+                                si 'xz' est introuvable sur le système.
+
+    Environnement :
+
+        TMPDIR                  Définit un répertoire temporaire spécifique au
+                                lieu de '/tmp' pour les étapes intermédiaires
+                                de l'empaquetage.
+
+
+    Rapports de bogues, suggestions, retours, questions, etc. sont à envoyer à
+    Seb (l'auteur et mainteneur ; la licence MIT/X11  se trouve en tête de
+    chaque script) à cette adresse : sbb CHEZ tuxfamily POINT org
 
 HELP
     exit $code
@@ -247,7 +246,7 @@
     local cmd="${1:-"xz"}"
     cmd=$(which "$cmd" 2>/dev/null) || cmd="$(which "gzip" 2>/dev/null | sed 's/zip$/zip -9/')"
     #        "-9" has no effect on xz and seems to (slightly) slow things ^^^^^^^^^^
-    [ "$cmd" ] || growl 1 "Can't find gzip in your PATH."
+    [ "$cmd" ] || growl 1 "'gzip' ne se trouve pas dans votre PATH."
     echo "$cmd -c"
 }
 
@@ -270,20 +269,20 @@
         -q|--quiet) exec 3>"/dev/null" ;;
         -r|--reset-permissions) reset=1 ;;
         -z|--gzip) zip="gzip" ;;
-        -*) growl 1 "\"$arg\": illegal option.";;
+        -*) growl 1 "\"$arg\": option incorrecte.";;
         *)  if [ ! "$source" ]; then
                 if [ ! -d "$arg" ]; then
-                    echo "The source is not a directory, trying to convert a Slackware(R) package..." >&3
+                    echo "La source n'est pas un répertoire : tentative de conversion d'un paquet Slackware(R)..." >&3
                     source="$(explode "$tmpdir" "$arg")" || exit 1
                 else
-                    source="$(cd "$arg" && pwd -P)" || growl 1 "Can't enter the \"$arg\" directory."
+                    source="$(cd "$arg" && pwd -P)" || growl 1 "Impossible d'accéder au répertoire \"$arg\"."
                 fi
             else
-                [ ! "$target" ] || growl 1 "Too many arguments."
+                [ ! "$target" ] || growl 1 "Trop d'arguments."
                 target="$arg"
-                arg="$(cd $(dirname "$arg") && pwd -P)" || growl 1 "No path to \"$target\"."
+                arg="$(cd $(dirname "$arg") && pwd -P)" || growl 1 "Aucun chemin vers \"$target\"."
                 if awk 'BEGIN{exit (index("'"$arg"'/","'"$source"'/") == 1) ? 0 : 1}'; then
-                    growl 1 "Can't generate the package into its source directory."
+                    growl 1 "Impossible de générer le paquet dans son répertoire source."
                 fi
                 target="$arg/$(basename "$target" .spack).spack"
             fi
@@ -291,14 +290,14 @@
 done
 [ "$source" ] || usage 1
 [ "$target" ] || target="$(echo "$source" | sed 's/\(\.tmp......\)\?$/.spack/')"
-echo "Generating the \"$target\" package..." >&3
-rm -fr "$target" || growl 1 "\"$target\" already exists and can't be removed."
-tmpdir=$(mktemp -d "$tmpdir/pkg-$$-XXXXXX" 2>/dev/null) || growl 1 "Can't create a secure temporary directory."
+echo "Création du paquet \"$target\"..." >&3
+rm -fr "$target" || growl 1 "\"$target\" existe déjà et ne peut pas être supprimé."
+tmpdir=$(mktemp -d "$tmpdir/pkg-$$-XXXXXX" 2>/dev/null) || growl 1 "Impossible de créer un répertoire temporaire sécurisé."
 cd "$source" || exit 1
 if [ $reset -ne 0 ]; then
-    echo "Forcing general root ownership..." >&3
+    echo "Appartenance des répertoires forcée à root..." >&3
     chown 0:0 -R .
-    echo "Forcing directory mode to 755 (rwx-r-x-r-x)..." >&3
+    echo "Droits des répertoires forcés à 755 (rwx-r-x-r-x)..." >&3
     # NOTE: POSIX minima allows 4096 arguments, but Linux generally goes over 100000.
     # Both cases seem to be enough to make some optimizations with "find".
     find . -type d -exec chmod 0755 {} +
@@ -309,7 +308,7 @@
 symrecord "$prepend" "$tmpdir"  >&3
 makefiles "$zip" "$target" "$tmpdir" >&3
 if [ $clean -ne 0 ]; then
-    echo "Removing the \"$source\" source directory..." >&3
+    echo "Suppression de répertoire source \"$source\"..." >&3
     rm -rf "$source"
 fi
 
