--- 0g.orig	2013-06-20 19:13:02.192056490 +0200
+++ 0g	2013-06-20 19:10:55.458049418 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 #
-# 0G (I) gestionnaire distant pour 0linux.
+# 0G (III) gestionnaire distant pour 0linux.
 #
 
 # Copyright (c) 2012 Sébastien Boillod <sbb at tuxfamily dot org>.
@@ -103,19 +103,27 @@
     $CURL "$db" | awk 'BEGIN { while ((getline <"'"$APPSCF"'")> 0) {
                                     if ($1 ~ /^@/) ALLOW[$1]
                                }
-                               close("'"$APPSCF"'"); delete ARGV[0];fd=0
+                               close("'"$APPSCF"'"); delete ARGV[0]
                                for (x in ARGV){
                                    x = ARGV[x]
                                    if (!(x in ALLOW)) {
                                    '$w'print "(W) "x": dépôt non-déclaré!">"/dev/stderr";continue
-                                       print x >>"'"$APPSCF"'"; fd=1
+                                    FORCE[x]
                                    }
                                    GET[x]
                                }
-                               if (fd > 0) close("'"$APPSCF"'");ARGC=split("",ARGV)
+                               ARGC=split("",ARGV)
                            }
                            ($1 in '$get'){delete '$get'[$1];$1="";print}
-                           END{for (x in '$get') print "(W) "x": dépôt introuvable.">"/dev/stderr"}' $1
+                           END{ for (x in '$get') {
+                                 print "(W) "x": dépôt introuvable.">"/dev/stderr"
+                                 delete FORCE[x]
+                                }
+                                for (x in FORCE)
+                                   print x >>"'"$APPSCF"'"
+                                if (x in FORCE)
+                                    close("'"$APPSCF"'")
+                            }' $1
 }
 
 upgrade () {
