--- a/xinitrc	2011-02-11 06:34:44.000000000 +0000
+++ b/xinitrc	2011-02-11 06:37:18.000000000 +0000
@@ -209,7 +209,11 @@
 
 # Run xfce4-session if installed
 if which xfce4-session >/dev/null 2>&1; then
-    xfce4-session
+    if which ck-launch-session >/dev/null 2>&1; then
+        ck-launch-session xfce4-session
+    else
+        xfce4-session
+    fi
 
     if test "$ssh_agent_kill_cmd"; then
         echo "running '$ssh_agent_kill_cmd'"
