--- spack-3.11.orig/src/spackadd	2014-07-05 16:51:54.281338469 +0000
+++ spack-3.11/src/spackadd	2014-07-05 22:06:41.595212697 +0000
@@ -81,7 +81,7 @@
     awk -v CATS="$cats" -v DIRS="$dirs" -v FILES="$files" \
         '(FNR == 1){if(NR>1)DELFILE=ARGV[1];STORE = 0}
          (FILENAME == DELFILE){next}
-         (STORE < 1){if ($0 ~ /^FILE LIST:/) STORE = length(DELFILE) > 0? 1 : 2; next }
+         (STORE < 1){if ($0 ~ /^FILE LIST:|^LISTE DES FICHIERS :/) STORE = length(DELFILE) > 0? 1 : 2; next }
          (STORE < 2){delete DEL[$0]; next}
          {DEL[$0]}
          END{ for(i in DEL){ if (i ~ /\/$/){print i>DIRS;continue}
@@ -119,7 +119,7 @@
     # Print the link list attached to the package, storing the others.
     # $f PACKAGE, KEPT LINK LIST, SCRIPTS
     local log="$POSTDIR/$1" kept="$2" post="$TMPDIR/post-install.sh"; shift 2
-    echo "FILE LIST:" >"$kept"
+    echo "LISTE DES FICHIERS :" >"$kept"
     if [ "$1" = "$POSTDIR/*" ]; then
         shift 1 # no more script left...
         [ -s "$post" ] || return 1 # nothing to save...
@@ -149,17 +149,17 @@
         exec 4>"$LOGDIR/$name"
     fi
     ( cd "$TMPDIR"
-        echo "PACKAGE NAME: $name" >&4
+        echo "NOM DU PAQUET : $name" >&4
         awk -v POST="$(du -k ./post-install.sh 2>/dev/null)" -v PRE="$(du -k ./pre-install.sh)" \
             '(NR==1){ZSIZE=$1;}(NR==2){SIZE=$1}
              END{RSIZE = 60 * (NR-2) / 512 # empty directory = 60bits
                 sub(/[^0-9]*$/,"",POST); sub(/[^0-9]*$/,"",PRE)
-                print "COMPRESSED SIZE: " int(ZSIZE/2) "K"
-                print "UNCOMPRESSED SIZE: " int((SIZE+RSIZE)/2-PRE+POST) "K"}' ./zsize ./size ./dirs >&4
-        echo "PACKAGE LOCATION: $path" >&4
-        echo "PACKAGE DESCRIPTION:" >&4
+                print "TAILLE COMPRESSÉE : " int(ZSIZE/2) " Kio"
+                print "TAILLE DÉCOMPRESSÉE : " int((SIZE+RSIZE)/2-PRE+POST) " Kio"}' ./zsize ./size ./dirs >&4
+        echo "EMPLACEMENT DU PAQUET : $path" >&4
+        echo "DESCRIPTION DU PAQUET :" >&4
         [ -r ./about.txt ] && awk '(L > 10){exit}/^'"$short"':/{L++;print;next}(L>0){exit}' ./about.txt >&4
-        echo "FILE LIST:" >&4
+        echo "LISTE DES FICHIERS :" >&4
         echo "./" >&4 # meet the Slackware spec.
         sort ./dirs ./files >&4 ) 
     exec 4>&- 
@@ -274,7 +274,7 @@
                 growl 0 "Impossible de supprimer le répertoire manquant \"$dir\"."
             fi
         done
-        [ $up -eq 0 ] || stamp=$(date "+-upgraded-%Y-%m-%d,%T")
+        [ $up -eq 0 ] || stamp=$(date "+-misajour-le-%d-%m-%Y-a-%T")
         [ $dry -gt 0 ] || mv -f "$LOGDIR/$1" "$RMLOGDIR/$1$stamp"
         [ $dry -gt 0 -o ! -f "$POSTDIR/$1" ] || mv -f "$POSTDIR/$1" "$RMPOSTDIR/$1$stamp"
         shift 1
@@ -342,7 +342,7 @@
 @ADDSpackadd installe ou met à niveau les paquets spécifiés, lesquels doivent
 @ADDrépondre aux spécifications de nommage originelles de Slackware :
 @ADD
-@ADD    « nom-version-architecture-décomptedesempaquetages »
+@ADD    « nom-version-architecture-révision »
 @ADD
 @ADDet porter l'extension « .spack ».
 @ADD'spackadd' tente d'abord de mettre à niveau l'installation des paquets
@@ -354,7 +354,7 @@
 @RM*spackrm - supprime un paquet du système.
 @RM
 @RM Spackrm supprime le ou les paquets spécifiés. On peut les spécifier avec
-@RM leur nom canonique : "nom-version-architecture-empaquetage",
+@RM leur nom canonique : "nom-version-architecture-révision",
 @RM ou leur simple nom de base, "nom".
 @RM Tous les paquets correspondants à la requête sont supprimés.
 @RM
